csrc/punica/punica_pybind.cpp

```diff
@@ -1,12 +1,15 @@
 #include <torch/extension.h>
-
+#include <git.h>
 #include "punica_ops.h"
 
 //====== pybind ======
 
 #define DEFINE_pybind(name) m.def(#name, &name, #name);
 
+std::string githash() { return std::string{git::CommitSHA1()}; }
```

Comment:
Why does the punica module need githash too?  I would think csrc/pybind.cpp and csrc/cpu/pybind.cpp were sufficient.

Reply:
I'm actually not sure if I should update punica or not. I see we create 3 .so files after build and it seems that we can load different .so based on machine setup. If not needed, I can update it to remove the one for punica.

Reply:
Updated the code after talking to Bill, punica is not needed here.